starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/DetailActivity.java

```diff
@@ -0,0 +1,38 @@
+package ly.generalassemb.drewmahrt.shoppinglistdetailview;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.TextView;
+
+public class DetailActivity extends AppCompatActivity {
+    public static final String ID_KEY = "idKey";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_detail);
+
+        int id = getIntent().getIntExtra(ID_KEY,-1);
+
+        if (id == -1) {
+            finish();
```

Comment:
I would suggest you show an error message to the user first before kicking them back to the first activity.